third_party/wabt/include/wabt/interp/jit/jit-fmath-inl.h

```diff
@@ -0,0 +1,231 @@
+}
+
+static sljit_f64 floatMin(sljit_f64 lhs, sljit_f64 rhs) {
+  if (WABT_UNLIKELY(std::isnan(lhs) || std::isnan(rhs))) {
+    return std::numeric_limits<double>::quiet_NaN();
+  } else if (WABT_UNLIKELY(lhs == 0 && rhs == 0)) {
+    return std::signbit(lhs) ? lhs : rhs;
+  } else {
+    return std::min(lhs, rhs);
+  }
+}
+
+static sljit_f32 floatMax(sljit_f32 lhs, sljit_f32 rhs) {
+  if (WABT_UNLIKELY(std::isnan(lhs) || std::isnan(rhs))) {
+    return std::numeric_limits<float>::quiet_NaN();
+  } else if (WABT_UNLIKELY(lhs == 0 && rhs == 0)) {
+    return std::signbit(lhs) ? rhs : lhs;
+  } else {
+    return std::max(lhs, rhs);
+  }
+}
+
+static sljit_f64 floatMax(sljit_f64 lhs, sljit_f64 rhs) {
+  if (WABT_UNLIKELY(std::isnan(lhs) || std::isnan(rhs))) {
+    return std::numeric_limits<double>::quiet_NaN();
+  } else if (WABT_UNLIKELY(lhs == 0 && rhs == 0)) {
+    return std::signbit(lhs) ? rhs : lhs;
+  } else {
+    return std::max(lhs, rhs);
+  }
+}
+
+static sljit_f32 floatCopySign(sljit_f32 lhs, sljit_f32 rhs) {
+  return std::copysign(lhs, rhs);
+}
+
+static sljit_f64 floatCopySign(sljit_f64 lhs, sljit_f64 rhs) {
+  return std::copysign(lhs, rhs);
+}
+
+static void emitFloatBinary(sljit_compiler* compiler, Instruction* instr) {
+  Operand* operands = instr->operands();
+  JITArg args[3];
+
+  for (unsigned int i = 0; i < 3; ++i) {
+    floatOperandToArg(compiler, operands + i, args[i], i);
+  }
+
+  sljit_s32 opcode;
+  f32Function2Param f32Func = nullptr;
+  f64Function2Param f64Func = nullptr;
+
+  switch (instr->opcode()) {
+    case Opcode::F32Add:
+      opcode = SLJIT_ADD_F32;
+      break;
+    case Opcode::F32Sub:
+      opcode = SLJIT_SUB_F32;
+      break;
+    case Opcode::F32Mul:
+      opcode = SLJIT_MUL_F32;
+      break;
+    case Opcode::F32Div:
+      opcode = SLJIT_DIV_F32;
+      break;
+    case Opcode::F32Max:
+      f32Func = floatMax;
+      break;
+    case Opcode::F32Min:
+      f32Func = floatMin;
+      break;
+    case Opcode::F32Copysign:
+      f32Func = floatCopySign;
+      break;
+    case Opcode::F64Add:
+      opcode = SLJIT_ADD_F64;
+      break;
+    case Opcode::F64Sub:
+      opcode = SLJIT_SUB_F64;
+      break;
+    case Opcode::F64Mul:
+      opcode = SLJIT_MUL_F64;
+      break;
+    case Opcode::F64Div:
+      opcode = SLJIT_DIV_F64;
+      break;
+    case Opcode::F64Max:
+      f64Func = floatMax;
+      break;
+    case Opcode::F64Min:
+      f64Func = floatMin;
+      break;
+    case Opcode::F64Copysign:
+      f64Func = floatCopySign;
+      break;
+    default:
+      WABT_UNREACHABLE;
+  }
+
+  if (f32Func) {
+    sljit_emit_fop1(compiler, SLJIT_MOV_F32, SLJIT_FR0, 0, args[0].arg,
+                    args[0].argw);
+    sljit_emit_fop1(compiler, SLJIT_MOV_F32, SLJIT_FR1, 0, args[1].arg,
+                    args[1].argw);
+    sljit_emit_icall(compiler, SLJIT_CALL, SLJIT_ARGS2(F32, F32, F32),
+                     SLJIT_IMM, GET_FUNC_ADDR(sljit_sw, f32Func));
+  } else if (f64Func) {
```

Comment:
What happens with the result? It seems nothing is done with FR0 after the return.
Please also check MOVE_TO_REG / MOVE_FROM_REG to reduce unnecessary moves.